Asset/documentdb-dotnet-todo-app/src/DocumentDBRepository.cs

```diff
@@ -81,15 +81,12 @@ public static async Task DeleteItemAsync(string id, string category)
 
         public static void Initialize()
         {
-            var factory = new MSITokenProviderFactory(new MSILoginInformation(MSIResourceType.AppService));
-            KeyVaultClient kvClient = new KeyVaultClient(new KeyVaultClient.AuthenticationCallback(async (authority, resource, scope) =>
-            {
-                return (await factory.Create(resource).GetAuthenticationHeaderAsync(default(CancellationToken))).Parameter;
-            }));
+            var credential = new DefaultAzureCredential();
             var vaultBaseUrl = Environment.GetEnvironmentVariable("AZURE_KEYVAULT_URI");
-            var endpoint = kvClient.GetSecretAsync(vaultBaseUrl, "azure-documentdb-uri").GetAwaiter().GetResult().Value;
-            var authKey = kvClient.GetSecretAsync(vaultBaseUrl, "azure-documentdb-key").GetAwaiter().GetResult().Value;
-            client = new DocumentClient(new Uri(endpoint), authKey);
+            SecretClient secretClient = new SecretClient(new Uri(vaultBaseUrl),credential);
+            var endpoint = secretClient.GetSecretAsync("azure-documentdb-uri").GetAwaiter().GetResult().Value;
+            var authKey = secretClient.GetSecretAsync("azure-documentdb-key").GetAwaiter().GetResult().Value;
```

Comment:
Just use the sync methods we not have, i.e. `GetSecret()`. These methods are synchronous all the way down the call stack and avoid potential problems with sync-over-async.

Reply:
Or, better, since you're waiting on other async methods here, can `Initialize` just be async? Is the caller async already (typical in ASP.NET, though more so in newer .NET versions)?

Reply:
@heaths  - Updated.